app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -356,23 +351,17 @@ function toField(
   const displayValue = valueType ? valueType.display_text : null
   const icon = valueType ? getValueTypeIcon(valueType.constructor) : null
 
-  const dataQuality =
+  const dataQualityMetrics =
     typeof props.data === 'object' && 'data_quality_pairs' in props.data ?
-      props.data.data_quality_pairs
+      props.data.data_quality_pairs.map((obj: Record<string, number[]>) => {
+        const key = Object.keys(obj)[0] as string
+        return { [key]: obj[key]?.[index!] ?? 0 }
+      })
       // eslint-disable-next-line camelcase
-    : { number_of_nothing: [], number_of_whitespace: [] }
-
-  const nothingIsNonZero =
-    index != null && dataQuality?.number_of_nothing ?
-      (dataQuality.number_of_nothing[index] ?? 0) > 0
-    : false
-
-  const whitespaceIsNonZero =
-    index != null && dataQuality?.number_of_nothing ?
-      (dataQuality.number_of_whitespace[index] ?? 0) > 0
-    : false
+    : [{ number_of_nothing: 0 }, { number_of_whitespace: 0 }]
```

Comment:
We were supposed to make visualization not know about kinds of metrics, but somehow this information still 'leaks' into this default fallback. Just in next statement, they are filtered out though because of the `0` values. Shouldn't this just be `[]`?

```suggestion
    : []
```